internal/search/codeownership/job.go

```
@@ -0,0 +1,154 @@
+package codeownership
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	otlog "github.com/opentracing/opentracing-go/log"
+
+	"github.com/sourcegraph/sourcegraph/internal/gitserver"
+	"github.com/sourcegraph/sourcegraph/internal/search"
+	"github.com/sourcegraph/sourcegraph/internal/search/job"
+	"github.com/sourcegraph/sourcegraph/internal/search/result"
+	"github.com/sourcegraph/sourcegraph/internal/search/streaming"
+	"github.com/sourcegraph/sourcegraph/internal/trace"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+
+	codeownerspb "github.com/sourcegraph/sourcegraph/internal/own/codeowners/proto"
+)
+
+func New(child job.Job, includeOwners, excludeOwners []string) job.Job {
+	return &codeownershipJob{
+		child:         child,
+		includeOwners: includeOwners,
+		excludeOwners: excludeOwners,
```

I kinda liked that previous approach where "noOwner" and "anyOwner" were explicit flags on this thing, because it means that never would we have a parsing error that returns an owner as `""` drastically change what this filter does. That said, I don't feel too strongly about it.

---

@eseliger are you suggesting to replace the empty case with using an explicit `noOwner` flag of some kind? if so, I think having `file:has.owner()` and `-file:has.owner()` as a filter makes sense- that's what I went to immediately when testing the flag.

that said- should we fail silently if we can't parse the owner?

---

yep, essentially what was undone here: https://github.com/sourcegraph/sourcegraph/pull/45973/commits/3f479b46312b5a42ca0be45456bb694debaf5811

---

heh no strong preference, if you don't like it don't do it :)